lib/web_ui/lib/src/engine/text/layout_service.dart

```diff
@@ -16,6 +16,19 @@ import 'paragraph.dart';
 import 'ruler.dart';
 import 'text_direction.dart';
 
+/// A single canvas2d context to use for all text measurements.
+final DomCanvasRenderingContext2D _context = _createTextMeasurementCanvas();
+
+DomCanvasRenderingContext2D _createTextMeasurementCanvas() {
+  final DomCanvasElement canvas = createDomCanvasElement();
+  // We don't use this canvas to draw anything, so let's make it as small as
+  // possible to save memory.
+  canvas
+    ..width = 0
+    ..height = 0;
```

Comment:
I think we should be able to test that we use a 0x0 canvas.

Reply:
Added a test that checks for number of canvases created as well as the size.